### PR TITLE
Issue218 plasma2d extraction iter

### PR DESCRIPTION
**Pull request by Didou09**
Main changes:
----------------
* Warnings when loading data from MultiIDSLoader fails for some data (in particular when using method to_Plasma2D()) is now just as informative but more concise and easier to read.  
* method MultiIDSLoader.to_Plasma2D() now more robust versus one-time-step-only data
* Introduced the basics for handling of rectangular meshes (R, Z) for interpolation via Plasma2D (nearest interpolation only)
* User has to define R and Z grid as '2dmeshR' and '2dmeshZ'

* Minor bug fixes in imas2tofu introduced by latest changes, detected when testing by hand, fixed

Issues:
-------
Fixes, in devel, issue #218

### CODE REVIEW

**Comment by Didou09** [comment]
Ready for merging when reviewed @lasofivec :-)

File: tofu/data/_core.py [hunk 1]
```diff
@@ -2418,71 +2418,176 @@ def _checkformat_dtrm(dtime=None, dradius=None, dmesh=None,
...
-                        if lc[0]:
-                            msg += "  Duplicate nodes: %s\n"%str(nnodes - nodesu.shape[0])
-                            msg += "    - nodes.shape: %s\n"%str(dd[dk][k0]['nodes'].shape)
-                            msg += "    - unique nodes.shape: %s\n"%str(nodesu.shape)
-                        if lc[1]:
-                            msg += "  Duplicate faces: %s\n"%str(nfaces - facesu.shape[0])
-                            msg += "    - faces.shape: %s\n"%str(dd[dk][k0]['faces'].shape)
-                            msg += "    - unique faces.shape: %s"%str(facesu.shape)
-                        raise Exception(msg)
+                    assert v0['type'] in ['rect', 'tri', 'quadtri'], v0['type']
```
Comment (lasofivec) [inline tofu/data/_core.py]: I thought you wanted to get rid of "assert" statements?

File: tofu/imas2tofu/_core.py [hunk 2]
```diff
@@ -1934,14 +1937,29 @@ def get_data_all(self, dsig=None, stack=True,
...
+                                                      stack=stack,
+                                                      isclose=isclose,
+                                                      flatocc=flatocc, nan=nan,
+                                                      pos=pos, warn=False)
+                if len(dfail[ids]) > 0:
+                    anyerror = True
             except Exception as err:
                 msg = "Could not get data from %s"%ids
                 warnings.warn(msg)
+        if anyerror:
```
Comment (lasofivec) [inline tofu/imas2tofu/_core.py]: shouldn't `anyerror` should be set to True if an Exception was found aswell ?

File: tofu/data/_core.py (hunk 1 above)
Comment (Didou09) [reply tofu/data/_core.py]: You're right, I replace it with if / raise

**Pull request merged**